### ISSUE TITLE
Train and validation losses are all "nan"

### ISSUE THREAD

**Issue by utahman** [keras-team/autokeras#1529]
### Bug Description
Using AutoModel simple regressor with MNIST data.
One very second Running Trial train and validation losses are "nan".
It seems that weights are not initialized correctly before fit.
Or maybe some other problem?
Easy to repro with the complete code below.
Must point out that I specifically not using image classifier or regressor because originally found an issue on AutoModel and non-image data. So MNIST data is used just as sample data for regression.

Epoch 1/1000
938/938 [==============================] - 15s 10ms/step - loss: nan - mean_squared_error: nan - val_loss: nan - val_mean_squared_error: nan

C:\Program Files\Python37\lib\site-packages\kerastuner\engine\metrics_tracking.py:92: RuntimeWarning: All-NaN axis encountered
  return np.nanmin(values)

Epoch 2/1000
938/938 [==============================] - 5s 6ms/step - loss: nan - mean_squared_error: nan - val_loss: nan - val_mean_squared_error: nan

Later there was this:

Epoch 9/1000
306/306 [==============================] - 38s 124ms/step - loss: 3576478020073272826134528.0000 - mean_squared_error: 3576478020073272826134528.0000 - val_loss: 3511374051458008265785344.0000 - val_mean_squared_error: 3511374051458008265785344.0000
Epoch 10/1000
306/306 [==============================] - 38s 124ms/step - loss: 3489985560567291014283264.0000 - mean_squared_error: 3489985560567291014283264.0000 - val_loss: 3426429100841584994287616.0000 - val_mean_squared_error: 3426429100841584994287616.0000
Epoch 11/1000
306/306 [==============================] - 38s 123ms/step - loss: 3405585894879488391512064.0000 - mean_squared_error: 3405585894879488391512064.0000 - val_loss: 3343582755593922335997952.0000 - val_mean_squared_error: 3343582755593922335997952.0000
Epoch 12/1000

Such losses should happen if weights initialized correctly

### Bug Reproduction
Code for reproducing the bug:

import autokeras as ak
from keras.datasets import mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train = x_train.reshape((60000,784)) / 256
y_train = y_train / 10
x_test = x_test.reshape((10000,784)) / 256
y_test = y_test / 10

amodel = ak.AutoModel(inputs=ak.StructuredDataInput(), outputs=ak.RegressionHead())
amodel.fit(x_train, y_train, batch_size=64, validation_data=[x_test, y_test])

### Expected Behavior
Data is simple and well known. Processing it should not cause losses being nan

### Version used
AutoKeras package 1.0.12

**Comment by haifeng-jin** [comment]
This is likely because of the search space.
The raw pixels are treated as structured data.
The search space of structured data tasks may not work for these inputs.

**Comment by utahman** [comment]
Is there some other configuration that can be used instead? It's just 60K samples with 784 features each. All perfectly scaled between zero and one. Same thing with targets. It could come from images or anything else. How AutoKeras can help with such simple arrangement of good scaled data?

**Comment by shun-lin** [comment]
If you use playground dataset that are meant to be regression like Boston house pricing does the same problem occurs?

**Comment by holinov** [comment]
> Is there some other configuration that can be used instead? It's just 60K samples with 784 features each. All perfectly scaled between zero and one. Same thing with targets. It could come from images or anything else. How AutoKeras can help with such simple arrangement of good scaled data?

you could try use `ak.Input()` instead of `ak.StructuredDataInput()`

**Comment by nalexus** [comment]
The same problem has occured to me on trial-by-trial basis with StructuredRegressor, i.e. with few trials val losses are calculated, but with others are not . It occured so after increasing batch size to smth like 254 (on batch_size=64 all val losses were NaNs).

**Comment by Anselmoo** [comment]
@haifeng-jin is there a chance for an early abort in case of constant production of `nan`?
thx
